src/transformers/integrations.py

```diff
@@ -1028,30 +1030,57 @@ def _stop_run_if_exists(self):
             del self._run
             self._run = None
 
-    def _reinitialize_run(self):
+    def _initialize_run(self, **additional_neptune_kwargs):
+        from neptune.new import init_run
+        from neptune.new.exceptions import NeptuneMissingProjectNameException, NeptuneMissingApiTokenException
+
+        self._stop_run_if_exists()
+
+        try:
+            self._run = init_run(**self._init_run_kwargs, **additional_neptune_kwargs)
+            self._run_id = self._run['sys/id'].fetch()
+        except (NeptuneMissingProjectNameException, NeptuneMissingApiTokenException) as e:
+            raise NeptuneCallback.MissingConfiguration() from e
+
+    def _ensure_run_with_monitoring(self):
         if self._initial_run is not None:
             self._run = self._initial_run
+            self._run_id = self._run['sys/id'].fetch()
             self._initial_run = None
```

Comment:
What if `initial_run` has monitoring disabled?

Reply:
That's fine :wink: